NAS-110793 / 12.0 / remove lxml and xml.etree from identify_freebsd

yocalebo opened the pull request:
- remove `lxml` and `xml` from `identify_freebsd` and use the built-in `geom.class_by_name.xml` attribute
- while here, remove unused `identify_linux` and `device_info_linux` modules (1 of which actually imports `lxml`)
- if a `{serial}` was requested to be identified, we were calling `geom.class_by_name.xml` twice for which I've made it so that we only call it once

bugclerk commented:
Jira URL: https://jira.ixsystems.com/browse/NAS-110793

themylogin commented on src/middlewared/middlewared/plugins/disk_/identify_freebsd.py:
```diff
@@ -68,32 +64,32 @@ def identifier_to_device(self, ident, disks=None):
+
+            # remove leading/trailing and more than single whitespace char(s)
+            _value = ' '.join(value.split())
+            search = xml.find(f'.//provider/config[ident = "{_value}"]/../../name')
```
Are we sure that XML document values are always whitespace-normalized?

yocalebo replied:
It's probably not, so I've fixed the logic.

rick-mesta commented:
@themylogin / @yocalebo : What's the status on this one, gents ? Are all issues resolved ?

yocalebo commented:
> @themylogin / @yocalebo : What's the status on this one, gents ? Are all issues resolved ?

I've addressed the review, just waiting on approval/re-review from @themylogin

rick-mesta commented:
Noice ! Thx @themylogin 😁
